packages/core/useElementVisibility/index.ts

```diff
@@ -19,40 +19,44 @@ export interface UseElementVisibilityOptions extends ConfigurableWindow {
  */
 export function useElementVisibility(
   element: MaybeComputedElementRef,
-  { window = defaultWindow, scrollTarget }: UseElementVisibilityOptions = {},
+  options: UseElementVisibilityOptions = {},
 ) {
-  const elementIsVisible = ref(false)
+  const { scrollTarget } = options
 
-  const testBounding = () => {
-    if (!window)
-      return
+  const elementBounding = reactive(useElementBounding(element))
+  const targetBounding = reactive(useElementBounding(scrollTarget))
+  const viewportSize = reactive(
+    useWindowSize({
+      // pass down the `configurableWindow`
+      ...options,
+      includeScrollbar: false,
+    }),
+  )
 
-    const document = window.document
-    const el = unrefElement(element)
-    if (!el) {
-      elementIsVisible.value = false
-    }
-    else {
-      const rect = el.getBoundingClientRect()
-      elementIsVisible.value = (
-        rect.top <= (window.innerHeight || document.documentElement.clientHeight)
-          && rect.left <= (window.innerWidth || document.documentElement.clientWidth)
-          && rect.bottom >= 0
-          && rect.right >= 0
-      )
+  const elementIsVisible = computed<boolean>(() => {
+    if (scrollTarget) {
+      // relative to the `scrollTarget`
+      return elementBounding.top <= targetBounding.bottom
+      && elementBounding.left <= targetBounding.right
+      && elementBounding.bottom >= targetBounding.top
+      && elementBounding.right >= targetBounding.left
     }
-  }
-
-  watch(
-    () => unrefElement(element),
-    () => testBounding(),
-    { immediate: true, flush: 'post' },
-  )
+    return (
+      // relative to the viewport/window
+      elementBounding.top <= viewportSize.height
+        && elementBounding.left <= viewportSize.width
+        && elementBounding.bottom >= 0
+        && elementBounding.right >= 0
+    )
+  })
 
-  if (window) {
-    useEventListener(scrollTarget || window, 'scroll', testBounding, {
-      capture: false, passive: true,
-    })
+  if (scrollTarget) {
```

Comment:
The check here seems to be not right. `scrollTarget` can be a template ref that is fulfilled later. We was checking for `window` for SSR.

Reply:
I thought `scrollTarget` can be `undefined` instead of `Ref<undefined>`, but now I checked `useEventLister` and it seems already SSR friendly, so I can just remove that check? (don't need to check the `window` either before?)